### DISCUSSION TITLE
How do I add a retention policy to a preview deployment?

### DISCUSSION THREAD

**Discussion by mastoj** [vercel/community#1035] [Idea]
The preview deployment is an amazing feature as I see it, but I would really like to see a retention policy configuration for it. I'm not to concerned when it comes to security because you can protect them, but I do see a risk of someone going to the wrong page and start doing things. If those things is against an old version there might be a risk that the system as a whole ends up in a wrong state. Yes, everyone should do upgrades in a expand contract kind of way so that shouldn't be much of an issue either, but everyone knows that we as human aren't perfect and will make mistakes.

What would I like to see is either one of two things or both:
* Being able to configure a retention policy for previews in the dashboard
* Have vercel to only deploy when you have a PR and then do a clean up when PR is merged.

**Comment by amyegan** [comment]
Thanks for reaching out @mastoj! This feature does not currently exist, but I know the team is exploring ways to clean up "expired" deployments automatically because you're not the only one to express interest in this kind of thing. I'm going to mark this as a feature suggestion!
